climpred/prediction.py

```diff
@@ -246,11 +253,16 @@ def compute_hindcast(
     for i in forecast.lead.values:
         if max_dof:
             forecast, reference = reduce_time_series(forecast, reference, i)
-        # take lead year i timeseries and convert to real time
+        # take lead year i timeseries and convert to real time based on temporal
+        # resolution of lead.
+        offset_args_tuple = get_lead_cftime_shift_args(
+            getattr(forecast['lead'], 'units'), i
```

Comment:
Is this the usual way of getting xarray attrs? I usually do ds['lead'].attrs['units']

Reply:
I'm not totally sure why we went this route. I think either works but for consistency we went with this everywhere. I think this looks kinda nice but I also see the logic in using native xarray.